contracts/staking/PoSValidatorManager.sol

```diff
@@ -130,4 +140,155 @@ abstract contract PoSValidatorManager is IPoSValidatorManager, ValidatorManager
 
     function _lock(uint256 value) internal virtual returns (uint256);
     function _unlock(uint256 value, address to) internal virtual;
+
+    function _initializeDelegatorRegistration(
+        bytes32 validationID,
+        address delegator,
+        uint64 weight
+    ) internal nonReentrant {
+        PoSValidatorManagerStorage storage $ = _getPoSValidatorManagerStorage();
+
+        // Ensure the validation period is active
+        Validator memory validator = _getValidator(validationID);
+        require(
+            validator.status == ValidatorStatus.Active, "ValidatorManager: Validator not active"
+        );
+
+        // Ensure the delegator is not already registered
+        require(
+            $._delegatorStakes[validationID][delegator].weight == 0,
+            "ValidatorManager: Delegator already registered"
+        );
+
+        _checkAndUpdateChurnTracker(weight);
+
+        // Submit the message to the Warp precompile.
+        bytes memory setValidatorWeightPayload = ValidatorMessages
+            .packSetSubnetValidatorWeightMessage(
+            validationID, _getAndIncrementNonce(validationID), validator.weight + weight
+        );
+        $._pendingRegisterDelegatorMessages[validationID][delegator] = setValidatorWeightPayload;
+        bytes32 messageID = WARP_MESSENGER.sendWarpMessage(setValidatorWeightPayload);
+
+        // Store the delegator information
+        $._delegatorStakes[validationID][delegator] = Delegator({
+            weight: weight,
+            startedAt: uint64(block.timestamp),
+            endedAt: 0,
+            status: ValidatorStatus.PendingAdded
```

Comment:
delegations are similar to validations, but seems strange to use the validator status enum

Reply:
Added a distinct `DelegatorStatus` enum that removes `Invalidated` as an option.